include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -113,102 +169,122 @@ struct __parallel_transform_reduce_small_submitter<__work_group_size, __iters_pe
     operator()(_ExecutionPolicy&& __exec, _Size __n, _ReduceOp __reduce_op, _TransformOp __transform_op,
                _InitType __init, _Ranges&&... __rngs) const
     {
-        auto __transform_pattern =
-            unseq_backend::transform_reduce_known<_ExecutionPolicy, __iters_per_work_item, _ReduceOp, _TransformOp>{
+        using _NoOpFunctor = unseq_backend::walk_n<_ExecutionPolicy, oneapi::dpl::__internal::__no_op>;
+        auto __transform_pattern1 =
+            unseq_backend::transform_reduce<_ExecutionPolicy, __iters_per_work_item1, _ReduceOp, _TransformOp>{
                 __reduce_op, _TransformOp{__transform_op}};
+        auto __transform_pattern2 =
+            unseq_backend::transform_reduce<_ExecutionPolicy, __iters_per_work_item2, _ReduceOp, _NoOpFunctor>{
+                __reduce_op, _NoOpFunctor{}};
         auto __reduce_pattern = unseq_backend::reduce_over_group<_ExecutionPolicy, _ReduceOp, _Tp>{__reduce_op};
 
-        const _Size __n_items = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __iters_per_work_item);
+        // number of buffer elements processed within workgroup
+        constexpr _Size __size_per_work_group = __iters_per_work_item1 * __work_group_size;
+        _Size __n_groups = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __size_per_work_group);
+        _Size __n_items = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __iters_per_work_item1);
 
-        sycl::buffer<_Tp> __res(sycl::range<1>(1));
+        sycl::buffer<_Tp> __temp((sycl::range<1>(__n_groups)));
 
         sycl::event __reduce_event = __exec.queue().submit([&, __n, __n_items](sycl::handler& __cgh) {
             oneapi::dpl::__ranges::__require_access(__cgh, __rngs...); // get an access to data under SYCL buffer
-            auto __res_acc = __res.template get_access<access_mode::write>(__cgh);
+            auto __temp_acc = __temp.template get_access<sycl::access_mode::write>(__cgh);
```

Comment:
I did not in my tests, but I changed it to provided additional information to the compiler. It might be useful on different platforms or in the future.